uk.co.compucorp.civicrm.hrleaveandabsences/api/v3/AbsencePeriod.php

```diff
@@ -44,3 +44,27 @@ function civicrm_api3_absence_period_delete($params) {
 function civicrm_api3_absence_period_get($params) {
   return _civicrm_api3_basic_get(_civicrm_api3_get_BAO(__FUNCTION__), $params);
 }
+
+
+/**
+ * AbsencePeriod.validPeriodExists API
+ *
+ * Checks whether a valid absence period exists or not.
+ * A valid absence period exists if a period exists that
+ * has its end date greater than or equal to today.
+ *
+ * @param array $params
+ *
+ * @return array API result descriptor
+ *
+ * @throws API_Exception
+ */
+function civicrm_api3_absence_period_validPeriodExists() {
```

Comment:
Was this not possible with the existing AbsencePeriod API? For me locally I can do a request for `AbsencePeriod.get` with `start_date` > today`?

Reply:
Thanks for the suggestion, didn't think of that earlier. 
`AbsencePeriod.get` with `end_date` >= today would be more appropriate though, because `start_date > today` won't catch for example a period with dates `2017-01-01 to 2017-12-31`, the aim is to get the closest period to the current date which also includes period containing the current date.